ledger/store/src/helpers/rocksdb/internal/mod.rs

```diff
@@ -91,6 +90,22 @@ pub struct RocksDB {
     pub(super) atomic_depth: Arc<AtomicUsize>,
     /// A flag indicating whether the atomic writes are currently paused.
     pub(super) atomic_writes_paused: Arc<AtomicBool>,
+    /// This is an optimization that avoids some allocations when querying the database.
+    pub(super) default_readopts: rocksdb::ReadOptions,
```

Comment:
Under the hood this does hold a foreign pointer which gets passed `get_pinned_opt`. Even though it doesn't get manipulated at all on the Rust side, is there certainty it would remain in the expected state for the lifetime of the Rocks object?

Reply:
While it would be quite weird if that were to happen, it's a valid concern; I've traced [the C code](https://github.com/facebook/rocksdb/blob/3f3045a40511d0e15a63b134e0c92ace49983fc1/db/c.cc#L6628-L6642) that uses the `ReadOptions`, though, and it doesn't seem to me that the object gets modified in any way, and the underlying pointer won't go away until a call to `rocksdb_readoptions_destroy` is made (which happens on `Drop`).

Reply:
@iamalwaysuncomfortable I've found a resource that states that `ReadOptions` may be reused freely: https://github.com/facebook/rocksdb/wiki/RocksDB-FAQ#basic-readwrite (the last Q in the linked section):
> Q: Can I reuse ReadOptions, WriteOptions, etc, across multiple threads?
> A: As long as they are const, you are free to reuse them.